### DISCUSSION TITLE
Getting pyhanko.sign.general.SigningError: Automatic signature size estimation is not available using the _ notation using version 0.18

### DISCUSSION THREAD

**Discussion by dimkar121** [MatthiasValvekens/pyHanko#268] [Q&A] [answered]
Hello,
I have upgraded to version 0.18 and I am getting the ``pyhanko.sign.general.SigningError: Automatic signature size estimation is not available exception`` although I have used the underscrore notation for signing_cert in the constructor of the Signer.
When I am switiching back to 0.17 , everything works fine.

best,
Dimitrios

**Comment by MatthiasValvekens** [comment]
Hi @dimkar121,


It's not entirely clear to me what you mean by "I have used the underscore notation for signing_cert in the constructor of the signer". Can you share the relevant piece of code?

I presume you already looked at the "breaking changes" section of the 0.18.x changelog? https://pyhanko.readthedocs.io/en/latest/changelog.html#release-0-18-0

**Reply by dimkar121** [reply]
Hi Matthias,
e.g.,
` self._signing_cert = signing_cert`
I have also tested my source with setters.

thanks in advance,
Dimitrios

**Reply by MatthiasValvekens** [reply]
Hi @dimkar121,


Right, but when/where are you doing this? This error is raised if the signing cert is not available at the time the signature container is allocated. Did you try stepping through your code with a debugger to see if this assignment actually happens before the call to `estimate_signature_container_size`?

Failing that, using the `bytes_reserved` param to `sign_pdf` or `digest_doc_for_signing` will also solve this issue since it short-circuits the estimation. Setting that value to something like 16k * 2 = 32k should be more than enough for standard-ish use cases.

**Reply by dimkar121** [reply]
Hi @MatthiasValvekens ,
I am doing the assignemnt to the signing_cert in the constructor of my Signer class using this simpe statement:
`self._signing_cert = signing_cert`
I have also used the `bytes_reserved` but then another exception was thrown : `pyhanko.pdf_utils.layout.LayoutError: Stamp text parameter 'signer' is missing`.
I am switching back to 0.17 leaving the code intact and it works flawlessly.

Any thoughts?

best,
Dimitrios

**Reply by MatthiasValvekens** [reply, marked as answer]
I see. Are you sure you're assigning `self._signing_cert = signing_cert` after the call to `super().__init__`? Because if not, the superclass is probably initialising it as `None`.

**Reply by dimkar121** [reply]
This was exactly the solution!
I had to invoke the setters after the `super().__init__`.
Thank you Matthias!
best,
Dimirios
